cluster/gce/util.sh

```diff
@@ -741,6 +746,73 @@ function create-master() {
   create-master-instance "${MASTER_RESERVED_IP}" &
 }
 
+# Adds master replica to etcd cluster.
+#
+# Assumed vars:
+#   REPLICA_NAME
+#   PROJECT
+#
+# $1: existing_master_name
+# $2: existing_manster_zone
+# $3: etcd client port
+# $4: etcd internal port
+# returns the result of ssh command which adds replica
+function add-replica-to-etcd() {
+  local -r existing_master_name="${1}"
+  local -r existing_manster_zone="${2}"
+  local -r client_port="${3}"
+  local -r internal_port="${4}"
+  gcloud compute ssh ${existing_master_name} \
+    --project "${PROJECT}" \
+    --zone "${existing_master_zone}" \
+    --command \
+      "curl localhost:${client_port}/v2/members -XPOST -H \"Content-Type: application/json\" -d '{\"peerURLs\":[\"http://${REPLICA_NAME}:${internal_port}\"]}'"
+  return $?
+}
+
+function replicate-master() {
+  set-replica-name
+
+  local existing_master=$(gcloud compute instances list \
+    --project "${PROJECT}" \
+    --regexp "$(get-replica-name-regexp)" \
+    --format "value(name,zone)" | head -n1)
+  local existing_master_name=$(echo "${existing_master}" | cut -f1)
+  local existing_master_zone=$(echo "${existing_master}" | cut -f2)
+
+  echo "Replicating existing master ${existing_master_zone}/${existing_master_name} as ${ZONE}/${REPLICA_NAME}"
+
+  # Before we do anything else, we should configure etcd to expect more replicas.
+  if ! add-replica-to-etcd ${existing_master_name} ${existing_master_zone} 4001 2380; then
+    echo "Failed to add master replica to etcd cluster."
+    return 1
+  fi
+  if ! add-replica-to-etcd ${existing_master_name} ${existing_master_zone} 4002 2381; then
+    echo "Failed to add master replica to etcd events cluster."
+    return 1
+  fi
```

Comment:
What I'm afraid about here is that if (after those commands) replicating master fails (e.g. because we weren't able to create PD, which potentially may happen), we end up in a state, where etcd still think that it contains more replicas.
That means that (potentially after more then one failure in a row), you will not longer have usable etcd cluster, because there will be less existing replicas than what quorum requires. This seems pretty dangerous.